src/dfx/src/lib/deps/mod.rs

```diff
@@ -260,7 +260,11 @@ pub fn get_pulled_service_candid_path(canister_id: &Principal) -> DfxResult<Path
 
 /// The path of the dir contains wasm and service.did.
 pub fn get_pulled_canister_dir(canister_id: &Principal) -> DfxResult<PathBuf> {
-    Ok(get_cache_root()?.join("pulled").join(canister_id.to_text()))
+    let p = get_cache_root()?;
+    if !p.exists() {
+        dfx_core::fs::create_dir_all(&p)?;
+    }
+    Ok(p.join("pulled").join(canister_id.to_text()))
```

Comment:
Given this method returns `some/dir/pulled/<canister id>/`, I'm not sure there is value in creating `some/dir/`.  The reason is that something else must be creating those subdirectories, and will have to create the parent directories if they don't exist.

```suggestion
    let p = get_cache_root()?;
    Ok(p.join("pulled").join(canister_id.to_text()))
```

That said:
- IF this method does create directories, it should create all of them.
- `if !p.exists() { create dir.. }` is known as `ensure_dir_exists(p)`

Reply:
wasn't sure here - I just wanted to mimic old behavior